crates/net/network/src/cache.rs

```diff
@@ -36,14 +44,33 @@ impl<T: Hash + Eq> LruCache<T> {
         false
     }
 
+    /// Same as [`Self::insert`] but returns a tuple, where the second index is the evicted value,
+    /// if one was evicted.
+    pub fn _insert_with_eviction_feedback(&mut self, entry: T) -> (bool, Option<T>) {
```

Comment:
I see, we can't express all 3 cases with a single option

* inserted new + not evicted
* inserted (already existing)
* inserted new + evicted

so this tuple makes sense.

I'd like to drop the _ here too.

I thought about a different name for a minute but I can't come up with anything.
but perhaps `insert_and_get_evicted`? fine with this though